lib/services/state/index.ts

```diff
@@ -54,15 +56,56 @@ class StateManager extends AbstractManager<{ utils: typeof utils }> implements I
     };
   }
 
+  parseDefaultValue(variableName: string, declare: any) {
+    const { defaultValue, isSystem, datatype } = declare;
+
+    let type = datatype;
+
+    if (isSystem) {
+      const builtInType = builtInVariableTypes.get(variableName);
+      if (!builtInType) {
+        throw new Error(`Received an invalid built-in variable with no defined type`);
+      }
+      type = builtInType;
+    }
+
+    switch (type) {
+      case VariableDatatype.BOOLEAN:
+        return defaultValue.toLowerCase() === 'true';
+      case VariableDatatype.DATE:
+        return new Date(defaultValue);
+      case VariableDatatype.NUMBER:
+        return parseFloat(defaultValue);
+      case VariableDatatype.TEXT:
+      case VariableDatatype.IMAGE:
+      case VariableDatatype.ANY:
+        return defaultValue;
+      default:
+        throw new Error(`Received unexpected variable type '${type}'`);
+    }
+  }
+
+  initializeFromCMSVariables(variables: Record<string, any>) {
+    return Object.entries(variables).reduce<Record<string, string | number | boolean | Date>>(
+      (acc, [name, declare]) => ({
+        ...acc,
+        [name]: this.parseDefaultValue(name, declare) ?? 0,
+      }),
+      {}
+    );
+  }
```

Comment:
better to use `Object.fromEntrie` to generate records, looks cleaner:
```suggestion
    return Object.fromEntries(Object.entries(variables).reduce<Record<string, string | number | boolean | Date>>(
      ([name, declare]) => [name, this.parseDefaultValue(name, declare) ?? 0]
    ));
  }
```